src/main/java/io/codelex/flightplanner/flight/response/FlightDataBaseService.java

```diff
@@ -0,0 +1,133 @@
+package io.codelex.flightplanner.flight.response;
+
+import io.codelex.flightplanner.flight.domain.Airport;
+import io.codelex.flightplanner.flight.domain.Flight;
+import io.codelex.flightplanner.flight.domain.PageResult;
+import io.codelex.flightplanner.flight.request.CreateFlightRequest;
+import io.codelex.flightplanner.flight.request.SearchFlightRequest;
+import jakarta.validation.Valid;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+
+
+public class FlightDataBaseService implements FlightService {
+
+
+    private final FlightDataBaseRepository flightDataBaseRepository;
+    private final AirportDataBaseRepository airportDataBaseRepository;
+
+    public FlightDataBaseService(FlightDataBaseRepository flightDataBaseRepository, AirportDataBaseRepository airportDataBaseRepository) {
+        this.flightDataBaseRepository = flightDataBaseRepository;
+        this.airportDataBaseRepository = airportDataBaseRepository;
+    }
+
+    private static LocalDateTime formatStringToLocalDateTime(String dateString) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+        if (dateString.length() == 10) {
+            dateString = dateString + " 00:00";
+        }
+
+        return LocalDateTime.parse(dateString, formatter);
+
+    }
+
+
+    @Override
+    public ListFlightResponse listFlights() {
+        return new ListFlightResponse(flightDataBaseRepository.findAll());
+    }
+
+    @Override
+    public synchronized boolean isFlightExists(CreateFlightRequest request) {
+        String fromCountry = request.getFrom().getCountry();
+        String fromCity = request.getFrom().getCity();
+        String fromAirport = request.getFrom().getAirport();
+        String toCountry = request.getTo().getCountry();
+        String toCity = request.getTo().getCity();
+        String toAirport = request.getTo().getAirport();
+        String carrier = request.getCarrier();
+        LocalDateTime arrivalTime = request.getArrivalTime();
+        LocalDateTime departureTime = request.getDepartureTime();
+        return flightDataBaseRepository.isFlightExist(fromCountry, fromCity, fromAirport, toCountry, toCity, toAirport, carrier, arrivalTime, departureTime);
+    }
+
+    @Override
+    public synchronized Flight createFlight(@Valid CreateFlightRequest request) {
+        if (!tripIsPossible(request)) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "This is a problem");
+        }
+        if (!dateIsCorrect(request)) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Date is not appropriated ");
+        }
+        if (!isFlightExists(request)) {
```

Comment:
Tie metožu nosaukumi isFlightExists un isAirportExists ir nedaudz nepareizi no angļu val. puses - tad labāk vienkārši to sauc par flightExists un airportExists.